src/ml/neural_net/mps_layer_instance_norm_data_loader.mm

```diff
@@ -0,0 +1,202 @@
+#include <ml/neural_net/mps_layer_instance_norm_data_loader.h>
+#include <ml/neural_net/mps_weight.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+@implementation TCMPSInstanceNormDataLoader
+
+- (instancetype) initWithParams:(NSString *)name
+                   gammaWeights:(float **)gammaWeights
+                    betaWeights:(float **)betaWeights
+          numberFeatureChannels:(NSUInteger)numberFeatureChannels
+                         styles:(NSUInteger)styles
+                         device:(id<MTLDevice> _Nonnull)dev 
+                      cmd_queue:(id<MTLCommandQueue> _Nonnull) cmd_q {
+  self = [self init];
+    
+  if (self) {
+    _name = name;
+    _numberOfFeatureChannels = numberFeatureChannels;
+
+    _styles = styles;
+    
+    _currentStyle = 0;  
+  
+    _gamma_weights = gammaWeights;
```

Comment:
How should we reason about the ownership of these weights? Where is the actual memory?

Reply:
solution, copy the weights